client/java/src/main/java/com/linecorp/centraldogma/client/AbstractWatcher.java

```diff
@@ -142,6 +188,12 @@ public void close() {
         if (currentWatchFuture != null && !currentWatchFuture.isDone()) {
             currentWatchFuture.cancel(false);
         }
+
+        if (meterRegistry != null) {
+            meterRegistry.remove(new Id(LATEST_REVISION_METER_NAME, tags, null, null, Type.GAUGE));
```

Comment:
Not sure if it's a good idea to remove this gauge since it takes long to remove it when there are a lot of meters in the map.
Anyway, how about creating the gauge in `start()` method and remove it here?
```java
this.gauge = Gauge.builder(LATEST_REVISION_METER_NAME, this, watcher -> {
    final Latest<T> latest = watcher.latest;
    if (latest == null) {
        return -1;
    } else {
        return latest.revision().major();
    }
}).tags(tags).register(meterRegistry);
...
meterRegistry.remove(gauge);
```

Reply:
> remove this gauge since it takes long to remove

The performance regression has been resolved, so removal operations will no longer take a long time. https://github.com/micrometer-metrics/micrometer/pull/5750#issuecomment-2550808550

Reply:
Just to be sure, I checked the implementation, and it seems that using `remove(Meter meter)` results in the same behavior.
https://github.com/micrometer-metrics/micrometer/blob/9b1d4042579ec0691c7b235478e7fe50ce5d8eab/micrometer-core/src/main/java/io/micrometer/core/instrument/MeterRegistry.java#L748

Reply:
> The performance regression has been resolved, so removal operations will no longer take a long time. https://github.com/micrometer-metrics/micrometer/pull/5750#issuecomment-2550808550

Thanks for the link. I didn't know they fixed it. 😉 

> it seems that using remove(Meter meter) results in the same behavior.

Yeah, it simply delegates the call to the `remove` method that takes a meter.
I left this comment while considering a scenario where `watcher.close()` is called without calling `watcher.start()`.
In that case, it would attempt to remove unregistered meters.
I understand this is unlikely in most cases, but I wanted to clarify it since modifying the code wouldn’t be difficult. 😉